File: libs/sdk-react-native/ios/BreezSDKListener.swift
import BreezSDK
import Foundation

class BreezSDKListener: EventListener {
    static let emitterName: String = "breezSdkEvent"
    
    func onEvent(e: BreezEvent) {
        if RNBreezSDK.hasListeners {
            RNBreezSDK.emitter.sendEvent(withName: BreezSDKListener.emitterName,
                                         body: BreezSDKMapper.dictionaryOf(breezEvent: e))
        }
    }
}

File: libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKListener.kt
package com.breezsdk

import breez_sdk.BreezEvent
import breez_sdk.EventListener
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezSDKListener(private val emitter: RCTDeviceEventEmitter): EventListener {
    companion object {
        var emitterName = "breezSdkEvent"
    }

    override fun onEvent(e: BreezEvent) {
        emitter.emit(emitterName, readableMapOf(e))
    }
}